client/src/components/Auth/PrivateRoute.tsx

```diff
@@ -0,0 +1,22 @@
+import { useLocation, Navigate } from "react-router-dom"
+
+interface IProps {
+    isAuthed: boolean,
+    children: JSX.Element
+}
+
+function PrivateRoute({ isAuthed, children }: IProps) {
+    let location = useLocation();
+
+    if (!isAuthed) {
+    // Redirect them to the /login page, but save the current location they were
+    // trying to go to when they were redirected. This allows us to send them
+    // along to that page after they login, which is a nicer user experience
+    // than dropping them off on the home page.
+    return <Navigate to="/login" state={{ from: location }} replace />;
+  }
+
+  return children;
```

Comment:
Since all the private routes will have the sidebar, better to put the `WithSideBar` component in the `PrivateRoute` rather than split into 2 different components. It would look something like this:
```suggestion
  return (
     <ContainerStyles>
            <SideBar />
            {Component}
     </ContainerStyles>
  );
```

Reply:
`HelmetWrapper` component also might be better in here rather than in each of the private pages

Reply:
> Since all the private routes will have the sidebar, better to put the `WithSideBar` component in the `PrivateRoute` rather than putting `WithSideBar` components in all the private pages. It would look something like this:

Wonderful suggestion :)

Reply:
> `HelmetWrapper` component also might be better in here rather than in each of the private pages

If we do this then we have to remember to use the `HelmetWrapper` for each of the public routes. Is this OK?

Reply:
Maybe something like a <Layout> component or whatever you want to call it which will include the helmet and other stuff we need in every page. I think we might need more than just helmet on every page

Reply:
I think we should wait until we need an abstraction for creating that personally. I think abstracting too early can make the wrapping component inflexible and hard to maintain. Maybe we all take a vote on this?

Reply:
@abdulqshabbir @cherylli 

Suggestion:
`PrivateRoutes` - have `HelmetWrapper`, `Sidebar` and routing to calendar, settings and search pages
`PublicRoutes` - have `HelmetWrapper` and routing to login and signup pages only

We have to check that if the user is logged in, they should not be able to access login and signup pages too, so I suggest `PublicRoutes` to check that if user is logged in, redirect them to calendar page (similar redirect pattern as `PrivateRoutes`)

Reply:
Or if the `HelmetWrapper` is the same for all pages, we better put it on the top level: App.tsx